### ISSUE TITLE
cannot remove tags

### ISSUE THREAD

**Issue by janbols** [Azure/azure-iot-sdk-java#251]
- **OS and version used:** Ubuntu 18.04

- **Java runtime used:** JDK8

- **SDK version used:** iot-service-client:1.13.3

# Description of the issue:

Currently the only way of updating tags on a device is by calling `com.microsoft.azure.sdk.iot.service.devicetwin.DeviceTwin#updateTwin` because `replaceTags` is not implemented yet.

When I call `updateTwin` with an updated `tagsMap` on the device, the updates are only persisted when adding or modifying existing tags. However, when removing a key in the map of tags or a key in one of the submaps in the tags, the updates on those keys are ignored.

For example:
existing tags map:
```json
{
    "tag1": "valueOfTag1,
    "tag2": {
        "tag2-1": "valueOfTag2-1",
        "tag2-2": "valueOfTag2-2"
    }
  }
```
Updating the above map of tags to the following ...
```json
{
    "tag1": "valueOfTag1,
    "tag2": {
        "tag2-1": "valueOfTag2-1"
    }
  }
```
will not remove `tag2-2`.

One workaround is to set the json object to some simple field and then set it back to a new json object like below: First update to ...
```json
{
    "tag1": "valueOfTag1,
    "tag2": false
  }
```
and then update to ...
```json
{
    "tag1": "valueOfTag1,
    "tag2": {
        "tag2-1": "valueOfTag2-1"
    }
  }
```

# Code sample exhibiting the issue:

# Console log of the issue:

# Need Support?
* Have a feature request for SDKs? Please post it on [User Voice](https://feedback.azure.com/forums/321918-azure-iot) to help us prioritize.
* Have a technical question? Ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-iot-hub) with tag “azure-iot-hub”
* Need Support? Every customer with an active Azure subscription has access to support with guaranteed response time.  Consider submitting a ticket and get assistance from Microsoft support team
* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C, Java, .NET, Node.js, Python).

**Comment by timtay-microsoft** [comment]
The way to remove a tag (or desired property) is to set it's value to null and then call updateTwin. See the example below:

```java
//Look at initial twin tags
System.out.println("Getting the Device twin");
twinClient.getTwin(device);
System.out.println(device);

//Update Twin Tags
Set<Pair> tags = new HashSet<Pair>();
tags.add(new Pair("<your tag to delete>", null));
device.setTags(tags);
twinClient.updateTwin(device);

//verify that it was deleted
twinClient.getTwin(device);
System.out.println(device);
```

However, this functionality is currently broken on our SDK. I'll have a fix checked in before the next release that will enable this again.

**Comment by janbols** [comment]
Great! Thanks.

**Comment by timtay-microsoft** [comment]
The fix for this issue has been checked in to master now. We are planning our next release for 6/15. Feel free to build from source to test it out in the meantime.

**Comment by timtay-microsoft** [comment]
The fix for the SDK side of this issue has been released as of Service Client 1.13.4

Because of that, I'm closing this issue. Feel free to re-open if you still face this issue.

@janbols  thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://www.surveymonkey.com/r/Q5PLZ3V)
